src/Network/Transport/TCP.hs

```diff
@@ -629,12 +663,189 @@ apiNewEndPoint transport =
     resolveMulticastGroupError =
       TransportError ResolveMulticastGroupUnsupported "Multicast not supported"
 
+-- | Specification of input queueing for a 'LocalEndPoint'
+data QPolicy prio = QPolicy {
+    -- | Determine a buffer constructor given an EndPointAddress.
+    --   Parametricity ensures the buffer is empty.
+    qpolicyBuffer :: forall t . EndPointAddress -> Buffer t
+    -- | Determine a priority for each EndPointAddress. Values which compare
+    --   lower give higher priority.
+  , qpolicyPriority :: EndPointAddress -> prio
```

Comment:
Updated.

I wonder whether this type should change. Perhaps the priority should be a function of the current priority queue as well `forall t . EndPointAddress -> PQ.MinPQueue prio t -> prio`?